File: behavioral/template/other_examples/use_template.java
public abstract class Airport {
    public void check() {
        checkPassport();
        checkBaggage();
        checkTicket();
        checkFlight();
        checkAirportLocation();
    }

    public void checkPassport() {
        System.out.println("check passport");
    }
    public void checkBaggage() {
        System.out.println("check baggage");
    }
    public void checkTicket() {
        System.out.println("check ticket");
    }
    public void checkFlight() {
        System.out.println("check flight");
    }
    public void checkAirportLocation() {
        System.out.println("check airport location");
    }

    //    abstract method
    public abstract void checkAirportLocation();
}

public class IncheonAirport extends Airport {
    public void checkAirportLocation() {
        System.out.println("check departure: Incheon");
    }
}

public class GimpoAirport extends Airport {
    public void checkAirportLocation() {
        System.out.println("check departure: Gimpo");
    }
}

File: behavioral/template/other_examples/use_template.py
from abc import ABCMeta, abstractmethod

class AirportMixin(metaclass=ABCMeta):
    def check(self):
        self.check_baggage()
        self.check_passport()
        self.check_ticket()
        self.check_flight()
        self.check_airport_location()

    def check_baggage(self):
        print("check baggage")

    def check_passport(self):
        print('check passport')

    def check_ticket(self):
        print('check ticket')

    def check_flight(self):
        print('check flight')

    @abstractmethod
    def check_airport_location(self):
        pass

class IncheonAirport(AirportMixin):
    def check_airport_location(self):
        print('check departure: Incheon')

class GimpoAirport(AirportMixin):
    def check_airport_location(self):
        print('check departure: Gimpo')
